### ISSUE TITLE
Neural network learns a constant function

### ISSUE THREAD

**Issue by Akella17** [pytorch/pytorch#17923]
Please find the ipython notebook in this link: https://drive.google.com/file/d/13Wtpd-Ur7tUO3mGpXQU2H08rNNyyqUyf/view?usp=sharing

The input to the model is 4D and output is 1D (regression problem). The code that I have shared is a simplification of the actual network while capturing the bug I am encountering.

**Comment by vishwakftw** [comment]
Could you please provide a small snippet or elaborate the issue you are facing so that the devs don't have to download and open?

If you think this is not a bug in the functioning of PyTorch, and is rather related to the usage, please consider posting to [the forums](https://discuss.pytorch.org).

**Comment by Akella17** [comment]
I haven't been able to isolate the reason for this behavior so I sent the entire code. The model learns to output a constant value (yes, it actually learns the constant value that minimizes the loss, i.e. mean of training outputs) for all the inputs (regression problem).

Intuitively, it feels like the weights are set to zero and the network is only learning biases. However, in practice, I have also noted that the weights are non zero.

**Comment by vishwakftw** [comment]
I think this would be better attended to in the forums, please open a post there. GitHub issues are primarily used for bug-tracking and feature requests.

**Comment by Akella17** [comment]
Sure thing. I have posted it there. Hope it gets resolved soon :)
Thanks for the quick response.
